### PR TITLE
MatQuill (Material Quill input), derived from QuillEditorBase

### PR DESCRIPTION
**Pull request by deavon**
- Adds an example QuillEditorBase-derived Component, `MatQuill` (which extends from `MatQuillBase`)
    - `MatQuillBase` implements all required `MatFormFieldControl` member variables & methods, for full interoperability with `@angular/components`
- Adds `MatQuill` variation to all `QuillEditorComponent` examples
- Some very basic layout refinements & sample style overrides

### CODE REVIEW

File: angular.json [hunk 1]
```diff
@@ -24,13 +24,13 @@
...
             "styles": [
-              "src/styles.css",
               "./node_modules/quill/dist/quill.core.css",
               "./node_modules/quill/dist/quill.bubble.css",
               "./node_modules/quill/dist/quill.snow.css",
               "./node_modules/quill-emoji/dist/quill-emoji.css",
               "./node_modules/quill-mention/dist/quill.mention.min.css",
-              "./node_modules/katex/dist/katex.min.css"
+              "./node_modules/katex/dist/katex.min.css",
+              "src/styles.scss"
```
Comment (deavon) [inline angular.json:33]: Moving `styles.scss` to the bottom to allow for cleaner override of any library styles

File: src/app/mat-quill/mat-quill-base.ts [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+		zone: NgZone,
+		config: QuillConfig,
+		service: QuillService
+	) {
+		super(
+			elementRef, domSanitizer, doc, platformId,
+			renderer, zone, config, service
+		)
+	}
+}
```
Comment (deavon) [inline src/app/mat-quill/mat-quill-base.ts:61]: Extends from `QuillEditorBase` before using `@angular/material/core` class mixins, also adding necessary constructor injection initializers.

File: src/app/mat-quill/mat-quill-base.ts [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+		return this.focused || !this.empty;
+	}
+
+	get value(): any
+	{
+		try {
+			return this.valueGetter(this.quillEditor, this.editorElem!);
+		} catch(e) {
+			return;
+		}
```
Comment (deavon) [inline src/app/mat-quill/mat-quill-base.ts:166]: `value` is needed immediately upon initialization, whereas `editorElem` is only ready `AfterViewInit`. This try / catch avoids more complicated solutions.

**Comment by KillerCodeMonkey** [comment]
@deavon nice thing!

could you please update ngx-quill to 10.0.2 in your pr? Your changes are in there :)

**Pull request merged**